src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_IAnalyzerHost.cs

```diff
@@ -24,10 +25,17 @@ public void AddAnalyzerAssembly(string analyzerAssemblyFullPath)
                 return;
             }
 
+            var fileChangeService = (IVsFileChangeEx)this.ServiceProvider.GetService(typeof(SVsFileChangeEx));
+            if (_visualStudioWorkspaceOpt == null)
+            {
+                // This can happen only in tests.
+                var testAnalyzer = new VisualStudioAnalyzer(analyzerAssemblyFullPath, fileChangeService, this.HostDiagnosticUpdateSource, this.Id, this.Workspace, loader: null, language: this.Language);
```

Comment:
If I understand this code correctly, you're solely doing this because you can't provide an IAnalyzerService? Why don't you provide a mocked one?